### PR TITLE
front: highlight selected train in GET

### LINKED ISSUE #291: GET: Highlight the selected train
Description
Quand un utilisateur sélectionne un train dans la liste des trains, le train sélectionné est mis en surbrillance dans le GET.

Acceptance criteria
- [ ] Le train sélectionné dans la liste des trains est mis en surbrillance dans le GET.

### PR DESCRIPTION
**Pull request by Uriel-Sautron**
close https://github.com/OpenRailAssociation/osrd-ui/issues/291

### CODE REVIEW

File: front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx [hunk 1]
```diff
@@ -172,7 +185,8 @@ const SimulationResultsV2 = ({ collapsedTimetable, spaceTimeData }: SimulationRe
             <div className="chart-container">
               <SpaceTimeChartWithManchette
                 operationalPoints={projectedOperationalPoints}
-                projectPathTrainResult={spaceTimeData}
+                projectPathTrainResult={formattedSpaceTimeDataToProjectPathTrainResult}
+                selectedProjection={trainUsedForProjectionSpaceTimeData?.id}
```
Comment (clarani) [inline front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx]: Ah oups we have a naming issue. We should highlight the selectedTrain, and not the train used for the projection. We should change the name of the field in the manchette too
```suggestion
                selectedProjection={selectedTrainSchedule?.id}
```

**Review by clarani — CHANGES_REQUESTED** [review]

File: front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx (hunk 1 above)
Comment (Uriel-Sautron) [reply front/src/applications/operationalStudies/views/v2/SimulationResultsV2.tsx]: Done

**Review by theocrsb — APPROVED** [review]
LGTM and tested

**Review by theocrsb — APPROVED** [review]

File: front/src/applications/operationalStudies/utils.ts [hunk 2]
```diff
@@ -290,3 +292,14 @@ export const isTooFast = (
   }
   return false;
 };
+
+export const formatSpaceTimeDataToProjectPathTrainResult = (
```
Comment (clarani) [inline front/src/applications/operationalStudies/utils.ts]: You don't need this function, you can simply replace trainName by name when you construct your SpaceTimeData ;)
The SpaceTimeData is built only at one place, in SimulationResultsV2. Change the type SpaceTimeData and then change the function getSpaceTimeChartData.
You will also need to change the function formatStdcmTrainIntoSpaceTimeData

Comment (Uriel-Sautron) [reply front/src/applications/operationalStudies/utils.ts]: Done thank you

**Review by clarani — APPROVED** [review]
LGTM & tested :white_check_mark:

**Pull request merged**